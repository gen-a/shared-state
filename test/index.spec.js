import React, {Component} from 'react';
import renderer from 'react-test-renderer';
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({adapter: new Adapter()});
import {shallow, mount} from 'enzyme';

// Component to be tested
import SharedState, {update} from '../src';

const Data = (props) => <div>{props.foo.bar}</div>;

class Input extends Component {
  constructor(props) {
    super(props);
    this.state = {value: props.foo.bar};
  }
  render() {
    const {foo, updateState} = this.props;
    return (
      <input
        type="text"
        value={this.state.value}
        onChange={(e) => updateState('foo', {...foo, bar: e.target.value})}
      />
    );
  }
}

describe('SharedState component testing...', () => {
    const dataComponent = renderer.create(<SharedState name="foo"><Data /></SharedState>);

    it('render correctly with empty state', () => {
      expect(dataComponent.toJSON()).toMatchSnapshot();
    });
    it('render correctly with updated state', () => {
      update('foo', {bar:2});
      expect(dataComponent.toJSON()).toMatchSnapshot();
    });
  it('change the value by input change event...', () => {
    const component = mount(<SharedState name="foo"><Input/></SharedState>);
    const domNode = component.find('input').getDOMNode();
    domNode.value = "33";
    component.find('input').simulate("change");
    expect(dataComponent.toJSON()).toMatchSnapshot();
  });
});
