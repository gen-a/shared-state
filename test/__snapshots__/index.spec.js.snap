// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedState component testing... change the value by input change event... 1`] = `
<div>
  33
</div>
`;

exports[`SharedState component testing... render correctly with empty state 1`] = `<div />`;

exports[`SharedState component testing... render correctly with updated state 1`] = `
<div>
  2
</div>
`;
